math/mathcore/src/TMathVectorized.cxx

```diff
@@ -331,7 +331,116 @@ Double_v TMath::LaplaceDistI(Double_v x, Double_t alpha, Double_t beta)
+   vecCore::Mask_v<Double_v> invalid_values = (p <= 0) || (p >= 1);
+
+   if(!vecCore::MaskEmpty(invalid_values)) {
+      Error("TMath::NormQuantile", "probability outside (0, 1)");
+      return 0;
+   }
+
+   Double_t  a0 = 3.3871328727963666080e0;
+   Double_t  a1 = 1.3314166789178437745e+2;
+   Double_t  a2 = 1.9715909503065514427e+3;
+   Double_t  a3 = 1.3731693765509461125e+4;
+   Double_t  a4 = 4.5921953931549871457e+4;
+   Double_t  a5 = 6.7265770927008700853e+4;
+   Double_t  a6 = 3.3430575583588128105e+4;
+   Double_t  a7 = 2.5090809287301226727e+3;
+   Double_t  b1 = 4.2313330701600911252e+1;
+   Double_t  b2 = 6.8718700749205790830e+2;
+   Double_t  b3 = 5.3941960214247511077e+3;
+   Double_t  b4 = 2.1213794301586595867e+4;
+   Double_t  b5 = 3.9307895800092710610e+4;
+   Double_t  b6 = 2.8729085735721942674e+4;
+   Double_t  b7 = 5.2264952788528545610e+3;
+   Double_t  c0 = 1.42343711074968357734e0;
+   Double_t  c1 = 4.63033784615654529590e0;
+   Double_t  c2 = 5.76949722146069140550e0;
+   Double_t  c3 = 3.64784832476320460504e0;
+   Double_t  c4 = 1.27045825245236838258e0;
+   Double_t  c5 = 2.41780725177450611770e-1;
+   Double_t  c6 = 2.27238449892691845833e-2;
+   Double_t  c7 = 7.74545014278341407640e-4;
+   Double_t  d1 = 2.05319162663775882187e0;
+   Double_t  d2 = 1.67638483018380384940e0;
+   Double_t  d3 = 6.89767334985100004550e-1;
+   Double_t  d4 = 1.48103976427480074590e-1;
+   Double_t  d5 = 1.51986665636164571966e-2;
+   Double_t  d6 = 5.47593808499534494600e-4;
+   Double_t  d7 = 1.05075007164441684324e-9;
+   Double_t  e0 = 6.65790464350110377720e0;
+   Double_t  e1 = 5.46378491116411436990e0;
+   Double_t  e2 = 1.78482653991729133580e0;
+   Double_t  e3 = 2.96560571828504891230e-1;
+   Double_t  e4 = 2.65321895265761230930e-2;
+   Double_t  e5 = 1.24266094738807843860e-3;
+   Double_t  e6 = 2.71155556874348757815e-5;
+   Double_t  e7 = 2.01033439929228813265e-7;
+   Double_t  f1 = 5.99832206555887937690e-1;
+   Double_t  f2 = 1.36929880922735805310e-1;
+   Double_t  f3 = 1.48753612908506148525e-2;
+   Double_t  f4 = 7.86869131145613259100e-4;
+   Double_t  f5 = 1.84631831751005468180e-5;
+   Double_t  f6 = 1.42151175831644588870e-7;
+   Double_t  f7 = 2.04426310338993978564e-15;
+
+   Double_t split1 = 0.425;
+   Double_t split2 = 5.;
+   Double_t konst1 = 0.180625;
+   Double_t konst2 = 1.6;
+
+   Double_v q, r, quantile;
+   q = p - 0.5;
+
+   vecCore::Mask_v<Double_v> values_lower_than_split1 = vecCore::math::Abs(q) < split1;
+
+   if (!vecCore::MaskEmpty(values_lower_than_split1)) {
+      vecCore::MaskedAssign<Double_v>(r, values_lower_than_split1, konst1 - q*q);
+      vecCore::MaskedAssign<Double_v>(quantile, values_lower_than_split1, q * (((((((a7 * r + a6) * r + a5) * r + a4) *
+                                      r + a3) * r + a2) * r + a1) * r + a0) / (((((((b7 * r + b6) * r + b5) * r + b4) *
+                                      r + b3) * r + b2) * r + b1) * r + 1.));
+   }
```

Comment:
The implementation would look a lot cleaner if these long expressions were put into their own functions.